src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/OppfolgingstilfelleService.kt

```diff
@@ -27,35 +28,42 @@ class OppfolgingstilfelleService(
     private val behandlendeEnhetClient: BehandlendeEnhetClient,
     private val pdlClient: PdlClient,
     private val syketilfelleClient: SyketilfelleClient,
+    private val oppfolgingstilfelleRetryProducer: OppfolgingstilfelleRetryProducer,
     private val producer: KafkaProducer<String, KOversikthendelsetilfelle>
 ) {
-    suspend fun receiveOppfolgingstilfeller(oppfolgingstilfellePeker: KOppfolgingstilfellePeker, callId: String = "") {
-        val aktor = AktorId(oppfolgingstilfellePeker.aktorId)
-
-        val fnr: String = aktorService.fodselsnummerForAktor(aktor, callId)
-            ?: return skipOppfolgingstilfelleWithMissingValue(MissingValue.FODSELSNUMMER)
-        val orgNummer = oppfolgingstilfellePeker.orgnummer
-        val organisasjonNavn = eregService.finnOrganisasjonsNavn(orgNummer, callId)
-
-        produce(oppfolgingstilfellePeker, fnr, organisasjonNavn, callId)
-    }
-
-    private fun skipOppfolgingstilfelleWithMissingValue(missingValue: MissingValue) {
-        when (missingValue) {
-            MissingValue.BEHANDLENDEENHET -> COUNT_OPPFOLGINGSTILFELLE_SKIPPED_BEHANDLENDEENHET.inc()
-            MissingValue.FODSELSNUMMER -> COUNT_OPPFOLGINGSTILFELLE_SKIPPED_FODSELSNUMMER.inc()
+    suspend fun receiveOppfolgingstilfelle(
+        aktorId: AktorId,
+        orgnummer: Virksomhetsnummer,
+        callId: String = ""
+    ) {
+        val isMissingMandatoryValue = processOppfolgingstilfelle(
+            aktorId,
+            orgnummer,
+            callId
+        ) != null
+        if (isMissingMandatoryValue) {
+            oppfolgingstilfelleRetryProducer.sendFirstOppfolgingstilfelleRetry(
+                aktorId,
+                orgnummer,
+                ""
+            )
+        } else {
+            LOG.info("Sent Oversikthendelsetilfelle on first attempt")
         }
     }
 
-    private suspend fun produce(
-        oppfolgingstilfellePeker: KOppfolgingstilfellePeker,
-        fnr: String,
-        organisasjonNavn: String,
+    suspend fun processOppfolgingstilfelle(
+        aktorId: AktorId,
+        orgnummer: Virksomhetsnummer,
         callId: String
-    ) {
+    ): MissingValue? {
```

Comment:
Brukes denne returverdien til noe, eller er det kun `missingValue != null`? I så fall kan det ev. byttes ut med en boolean?

Reply:
Det jeg tenkte var at om det er boolean, så klarer ikke kallende service å differensiere mellom hvorfor det ble feil.
La ikke inn noe nå, men tenkte kan være gunstig å abstrahere ut dette slik at kallende service kan ha ulik oppførsel avhengig av hvilket eksternt kall som feilet. Bruker det ikke nå, så kan jo evt. utsette det og fjerne det nå i denne PR'en.

Kunne evt. også returnert et objekt med en boolean for om det er vellykket og evt. hva som feilet.

Reply:
Endret til Boolean.